tck/src/test/java/org/reactivestreams/tck/IdentityProcessorVerificationTest.java

```diff
@@ -115,6 +127,145 @@ public void required_spec104_mustCallOnErrorOnAllItsSubscribersIfItEncountersANo
     }, "Did not receive expected error on downstream within " + DEFAULT_TIMEOUT_MILLIS);
   }
 
+  @Test
+  public void required_spec104_mustCallOnErrorOnAllItsSubscribersIfItEncountersANonRecoverableError_shouldAllowSignalingElementAfterBothDownstreamsDemand() throws Throwable {
+    final TestEnvironment env = newTestEnvironment();
+    new IdentityProcessorVerification<Integer>(env, DEFAULT_TIMEOUT_MILLIS) {
+      @Override
+      public Processor<Integer, Integer> createIdentityProcessor(int bufferSize) { // knowingly ignoring buffer size, acting as-if 0
+        return new Processor<Integer, Integer>() {
+
+          private volatile Subscription upstreamSubscription;
+
+          private final CopyOnWriteArrayList<MySubscription> subs = new CopyOnWriteArrayList<MySubscription>();
+          private final CopyOnWriteArrayList<Subscriber<? super Integer>> subscribers = new CopyOnWriteArrayList<Subscriber<? super Integer>>();
+          private final AtomicLong demand1 = new AtomicLong();
+          private final AtomicLong demand2 = new AtomicLong();
+          private final CountDownLatch awaitLatch = new CountDownLatch(2); // to know when both subscribers have signalled demand
+
+          @Override
+          public void subscribe(final Subscriber<? super Integer> s) {
+            int subscriberCount = subs.size();
+            switch (subscriberCount) {
+              case 0:
+                s.onSubscribe(createSubscription(awaitLatch, s, demand1));
```

Comment:
The problem with this is that the `Subscriber` will become visible to the `onNext` before `onSubscribe` is called and may run in arbitrary order in respect to each other or even concurrently with each other.